File: genetic-packer/src/main/java/genetic/packer/evolution/elitism/impl/PromoterImpl.java
package genetic.packer.evolution.elitism.impl;

import genetic.api.elitism.EliteAcknowledged;
import genetic.api.elitism.Promoter;
import genetic.api.individual.impl.Elite;
import genetic.api.individual.impl.RatedIndividual;

/**
 * @author piotr.larysz
 */
public class PromoterImpl implements Promoter, EliteAcknowledged {

    @Override
    public <M1 extends Comparable<M1>, M2> Elite<M1, M2> acknowledge(Elite<M1, M2> elite) {
        return elite;
    }

    @Override
    public <M1 extends Comparable<M1>, M2> Elite<M1, M2> apply(RatedIndividual<M1, M2> ratedIndividual) {
        return new Elite<>(ratedIndividual);
    }
}

File: genetic-evaluators/src/main/java/genetic/evaluator/elitism/impl/PromoterImpl.kt
package genetic.evaluator.elitism.impl

import genetic.api.elitism.EliteAcknowledged
import genetic.api.elitism.Promoter
import genetic.api.individual.impl.Elite
import genetic.api.individual.impl.RatedIndividual

class PromoterImpl : Promoter, EliteAcknowledged {

  fun <T> acknowledge(elite: Elite<T>) = elite

  override fun <T> invoke(ratedIndividual: RatedIndividual<T>) = Elite(ratedIndividual)
}